### PR TITLE
Add support for OSB parameter schemas

### LINKED ISSUE #490: Implement Open Service Broker parameter schemas
This issue is to implement parameter schemas in the Open Service Broker APIs as defined in:

- https://github.com/openservicebrokerapi/servicebroker/issues/59#issuecomment-282069469
- https://github.com/avade/servicebroker/tree/method-attached-schemas/_spec.md#catalog-mgmt

This means:

1.  Updating our internal OSB API client to handle the new fields
2.  Updating our API to incorporate the information 
3.  Determine what type of validations to add:
  1.  Likely we will need an admission controller to enforce validatons on Parameters for `Instance` and `Binding` versus their `ServiceClass` and `ServicePlan`
  2.  Should there be a way to bypass those validations?
  3.  Should extra parameters be allowed which are not explicitly mentioned in the schema?

I'm adding this to MVP-2, but it might realistically be a 'Later' item.

### PR DESCRIPTION
**Pull request by pmorie**
Closes #490.

I'm posting this early to see how people feel about the changes to our API; I'll finish the rest when I have cause to believe I won't have to re-do the API changes.

Believe it or not I have gone through a couple iterations before posting this that I didn't like.

### CODE REVIEW

**Comment by arschles** [comment]
@pmorie changes look ok, but I'd like to see what the controller does with these, so I have a bigger picture. Ping me when you want another review

**Comment by pmorie** [comment]
@arschles the controller will do nothing except transform them from the appropriate buckets in the OSB API.  These are basically there for people to build UIs around at this point; we can just treat them like blobs.

**Comment by pmorie** [comment]
Still needs:

- godoc
- test coverage
- control flags?

**Comment by vaikas-google** [comment]
As per discussions in the F2F the proposal being investigated is that the schemas are going to be on the ServiceInstance/ServiceBinding, and not below that in separate methods (like Create / Update, etc.) So, let's hold off on this a little bit.

**Comment by pmorie** [comment]
@vaikas-google 

> As per discussions in the Open Service Broker F2F the proposal being investigated is that the schemas are going to be on the ServiceInstance/ServiceBinding, and not below that in separate methods (like Create / Update, etc.) So, let's hold off on this a little bit.

This implements what is in the proposal here: https://github.com/openservicebrokerapi/servicebroker/pull/165 - did something change during the week?

**Comment by pmorie** [comment]
I have a business need to get something in to carry schema information ASAP, so unless the proposal has been redrafted, I would like for this to be in the next release.

**Comment by pmorie** [comment]
It looks based on this comment like Matt thinks the proposal should stay the way it is: https://github.com/openservicebrokerapi/servicebroker/pull/165#issuecomment-303367440

File: pkg/brokerapi/schemas.go [hunk 1]
```diff
@@ -16,16 +16,16 @@ limitations under the License.
...
 
-// Schemas represents a broker's schemas for both service instances and service
-// bindings
+// Schemas represents a plan's schemas for service instance and binding create
+// and update.
 type Schemas struct {
-	Instance Schema `json:"instance"`
-	Binding  Schema `json:"binding"`
+	ServiceInstances *Schema `json:"service_instances,omitempty"`
+	ServiceBindings  *Schema `json:"service_bindings,omitempty"`
```
Comment (arschles) [inline pkg/brokerapi/schemas.go]: this changes the wire protocol for communicating with brokers, no?

Comment (pmorie) [inline pkg/brokerapi/schemas.go]: It changes the application-layer messages, but they were not correct in terms of the actual proposed change to begin with.  They are correct after this PR.

**Comment by vaikas-google** [comment]
No tests?

**Comment by pmorie** [comment]
My bad - this has been pending for long enough I had forgotten that I had a couple tests to add.

**Pull request merged**